### PR TITLE
WIP: Add import/install example

### PR DESCRIPTION
**Pull request by antevens**

### CODE REVIEW

File: import_install_example.sh [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+# shellcheck disable=SC2120,SC2119
+function install_lib {
+    lib_path="${1:-/usr/local/bin/shtdlib.sh}"
+    lib_name="$(basename "${lib_path}")"
+    tmp_path="$(mktemp)"
+
+    echo "Installing library ${lib_name} to ${lib_path}"
+    original_umask="$(umask)"
+    umask 0022
+    download_lib "${tmp_path}"
```
Comment (hrpatel) [inline import_install_example.sh]: Add `... || return 1` if the call to `download_lib` fails. No point in continuing...

File: import_install_example.sh [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+# Library install function, optionallly accepts a URL and a full path/name
+# shellcheck disable=SC2120,SC2119
+function install_lib {
+    lib_path="${1:-/usr/local/bin/shtdlib.sh}"
+    lib_name="$(basename "${lib_path}")"
+    tmp_path="$(mktemp)"
+
+    echo "Installing library ${lib_name} to ${lib_path}"
+    original_umask="$(umask)"
```
[the remark this answers is not in the document]
Comment (antevens) [reply import_install_example.sh]: I had to double check but masks are always inherited yes.

File: import_install_example.sh (hunk 1 above)
Comment (antevens) [reply import_install_example.sh]: Not true, we still want to fix the umask so I'm going to fix this using the umask decorator :)

Comment (hrpatel) [reply import_install_example.sh]: Good point

File: import_install_example.sh [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+# Library import function, accepts one optional parameter, name of the file to import
+# shellcheck disable=SC2120,SC2119
+function import_lib {
+    lib_name="${1:-'shtdlib.sh'}"
+    full_path="$(readlink -f "${BASH_SOURCE[0]}" 2> /dev/null || realpath "${BASH_SOURCE[0]}" 2> /dev/null || greadlink -f "${BASH_SOURCE[1]}" 2> /dev/null:-"${0}")"
+    # Search current dir and walk down to see if we can find the library in a
+    # parent directory or sub directories of parent directories named lib/bin
+    while true; do
+        pref_pattern=( "${full_path}/${lib_name}" "${full_path}/lib/${lib_name}" "${full_path}/lib/${lib_name}" )
```
[the remark this answers is not in the document]
Comment (antevens) [reply import_install_example.sh]: My mistake, was supposed to be lib and bin but got muddled along the way.

**Pull request merged**